lib/frodata/middleware/raise_error.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module FrOData
+  class Middleware::RaiseError < Faraday::Response::Middleware
+    def on_complete(env)
+      @env = env
+
+      case env[:status]
+      when 300
+        raise Faraday::Error::ClientError.new("300: The external ID provided matches " \
+                                              "more than one record",
+                                              response_values)
+      when 401
+        raise FrOData::UnauthorizedError, message
+      when 404
+        raise Faraday::Error::ResourceNotFound, message
```

Comment:
Consider wrapping these Faraday errors with FrOData errors, so the internal implementation doesn't leak out.